developer-docs-site/docs/indexer/legacy/migration.md

```diff
@@ -0,0 +1,121 @@
+
+```yaml
+health_check_port: 8084
+server_config:
+  processor_name: default_processor
+  postgres_connection_string: postgresql://postgres:@localhost:5432/indexer
+  indexer_grpc_data_service_address: 127.0.0.1:50051
+  indexer_grpc_http2_ping_interval_in_secs: 60
+  indexer_grpc_http2_ping_timeout_in_secs: 10
+  auth_token: AUTH_TOKEN
+```
+
+To connect the processor to the Transaction Stream Service, you need to set the URL for `indexer_grpc_data_service_address`. Choose one of the following options.
+
+### Option a: Connect to Labs-provided Transaction Stream Service
+
+By using the Labs-provided Transaction Stream Service, you no longer have to run an archival node. Instructions to connect to Labs-provided Transaction Stream can be found [here](https://aptos.dev/indexer/txn-stream/labs-hosted).
+
+### Option b: Run a Self-Hosted Transaction Stream Service
+
+If you choose to, you can run a self-hosted instance of the Transaction Stream Service and connect your processors to it. Instructions to run a Self-Hosted Transaction Stream can be found [here](https://aptos.dev/indexer/txn-stream/self-hosted).
+
+## 3. (Optional) Migrate custom processors to Transaction Stream Service
+
+If you have custom processors written with the old indexer, we highly recommend starting from scratch and using a new database than what your legacy indexer is already writing to. Using a new database ensures that all your custom database migrations will be applied during this migration.
+
+### a. Migrate custom table schemas
+
+Migrate your custom schemas by copying over each of your custom migrations to the [`migrations`](https://github.com/aptos-labs/aptos-indexer-processors/tree/main/rust/processor/migrations) folder.
+
+### b. Migrate custom processors code
+
+Migrate the code by copying over your custom processors to the [`processors`](https://github.com/aptos-labs/aptos-indexer-processors/tree/main/rust/processor) folder and any relevant custom models to the [`models`](https://github.com/aptos-labs/aptos-indexer-processors/tree/main/rust/processor/src/models) folder. Integrate the custom processors with the rest of the code by adding them to the following Rust code files.
+
+[`mod.rs`](https://github.com/aptos-labs/aptos-indexer-processors/blob/main/rust/processor/src/processors/mod.rs)
+
+```
+pub enum Processor {
+    ...
+    CoinProcessor,
+    ...
+}
+
+impl Processor {
+    ...
+    COIN_PROCESSOR_NAME => Self::CoinProcessor,
+    ...
+}
+```
+
+[`worker.rs`](https://github.com/aptos-labs/aptos-indexer-processors/blob/main/rust/processor/src/worker.rs)
+
+```
+Processor::CoinProcessor => {
+    Arc::new(CoinTransactionProcessor::new(self.db_pool.clone()))
+},
+```
+
+## 4. Backfill Postgres database with Diesel
+
+Even though the new processors have the same Postgres schemas as the old ones, we recommend to do a complete backfill (ideally writing to a new DB altogether) because some fields are a bit different as a result of the protobuf conversion.
```

Comment:
```suggestion
Even though the new processors have the same Postgres schemas as the old ones, we recommend you do a complete backfill (ideally writing to a new DB altogether) because some fields are a bit different as a result of the protobuf conversion.
```